File: DraftRightMobile/ios/DraftRightKeyboardCore/Sources/DraftRightKeyboardCore/Composer/PassthroughComposer.swift
import Foundation

/// No-op composer for languages that type directly with no composition
/// (English and the other Latin packs). Every key/backspace passes through to
/// the input field unchanged; there is no composing buffer.
///
/// Lets every LanguagePack return a real Composer (uniform pipeline) instead of
/// nil. Behaviour matches the previous nil path (KeyboardController maps
/// passThrough → commit char / deleteOne).
public final class PassthroughComposer: Composer {
    public init() {}
    public func onKey(_ char: Character) -> ComposeResult { .passThrough }
    public func onBackspace() -> ComposeResult { .passThrough }
    public func reset() {}
    public func currentComposingText() -> String { "" }
}

File: DraftRightMobile/android/app/src/main/kotlin/com/draftright/keyboard/composer/PassthroughComposer.kt
package com.draftright.keyboard.composer

import com.draftright.keyboard.ComposeResult
import com.draftright.keyboard.Composer

/**
 * No-op composer for languages that type directly with no composition
 * (English and the other Latin packs). Every key/backspace passes through to
 * the input field unchanged; there is no composing buffer.
 *
 * Lets every LanguagePack return a real Composer (uniform pipeline) instead of
 * null — the keyboard never special-cases "no composer". Behaviour is identical
 * to the previous null path (KeyboardController maps PassThrough → Commit char /
 * DeleteOne).
 */
class PassthroughComposer : Composer {
    override fun onKey(char: Char): ComposeResult = ComposeResult.PassThrough
    override fun onBackspace(): ComposeResult = ComposeResult.PassThrough
    override fun reset() {}
    override fun currentComposingText(): String = ""
}
